src/main/java/seedu/address/model/ModuleSystem.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.UniqueModuleList;
+
+/**
+ * Wraps all data at the module-system level
+ * Duplicates are not allowed (by .isSameModule comparison)
+ */
+public class ModuleSystem implements ReadOnlyModuleSystem {
+
+    private final UniqueModuleList modules;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        modules = new UniqueModuleList();
+    }
```

Comment:
Sorry, would this have the same effect?

```suggestion
    private final UniqueModuleList modules = new UniqueModuleList();
```

Reply:
I followed the format of the original AB3 in case it matters

Reply:
I think it would be better to use Fergus' suggestion, non-static blocks might be "uncommon" to the future testers when we get to the evaluation phase.